src/launcher/x64dbg_launcher.cpp

```diff
@@ -432,6 +435,42 @@ static void EnableHiDPI()
     }
 }
 
+static std::wstring escape(std::wstring cmdline)
+{
+    StringUtils::ReplaceAll(cmdline, L"\\", L"\\\\");
+    StringUtils::ReplaceAll(cmdline, L"\"", L"\\\"");
+    return cmdline;
+}
+
+class CommandlineArguments : public ArgumentParser
+{
+public:
+    std::string filename;
+    std::vector<std::string> extraCmdline;
+    std::string pid;
+    std::string tid;
+    std::string event;
+    std::string command;
+    std::string commandFile;
+    bool help = false;
+
+    CommandlineArguments() : ArgumentParser("x64dbg")
+    {
+        addPositional("filename", filename, "Filename of program to debug");
+        addExtra(extraCmdline);

```

Comment:
This is a bit problematic, because it changes the command line interface of the launcher. This interface is documented and necessary for supporting GFlags and the shell integration, which simply pass `x96dbg.exe my.exe arg1 arg2`. I guess I could add support for extra arguments without a `--` delimiter, but perhaps it's better to leave the launcher alone for now?

Reply:
Hm, yes, maybe. It'd be easier to configure if you only need one path to the x96dbg.exe instead of two for the x32 and x64 ones. Or are the paths well defined relative to the x96dbg?

Reply:
Yeah they are always in the same spot relative to x96dbg. Also eventually I plan to completely deprecate x32dbg and implement proper wow64 support. Another idea could be to handle the transition to the right debugger executable directly in x64dbg.exe during command line parser.

Reply:
Yeah I think the command line interface of the launcher has to stay the same, just the final command line constructed for x64dbg has to be in the new format.

For the commands I think calling `scriptcmdexec` (on the command thread) instead of `DbgCmdExec` should mostly do the trick. Currently the machinery related to programmatically execute scripts is not that good, but doing this should work :tm:

```c++
String command = "scriptcmd ";
command += args.command;
DbgCmdExec(command.c_str());
```

For the script file it should also reuse the existing initialization script functionality, similar logic already exists in `cbInitializationScriptThread`. The question is what if there is already an initialization script, should the command line override it or run before/after?

Reply:
Another behavior could be to make the command line specify the initialization script, so if you restart the executable it will execute the same script automatically.